libraries/AP_Generator/AP_RichenPower.h

```diff
@@ -0,0 +1,190 @@
+// -*- tab-width: 4; Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+#pragma once
+
+#include <AP_Common/AP_Common.h>
+#include <AP_HAL/AP_HAL.h>
+#include <GCS_MAVLink/GCS.h>
+#include <SRV_Channel/SRV_Channel.h>
+
+#include <stdint.h>
+#include <stdio.h>
+
+#ifndef RICHENPOWER_ENABLED
+#define RICHENPOWER_ENABLED !HAL_MINIMIZE_FEATURES
+#endif
+
+#if RICHENPOWER_ENABLED
```

Comment:
I think we should make this GENERATOR_ENABLED instead of the more specific RICHENPOWER_ENABLED